src/gurt/fault_inject.c

```diff
@@ -534,6 +533,9 @@ d_fault_inject_init(void)
 	}
 	D_RWLOCK_UNLOCK(&d_fi_gdata.dfg_rwlock);
 
+	/* Enable fault inject by default for dev mode */
+	d_fault_inject = 1;
```

Comment:
the "FAULT_INJECTION" was set for non-release (dev/debug) build, not quite understand why need this change?

Reply:
I was wondering the same.

I do remember that in one instance I had to write a empty config file to enable fi otherwise I couldn't enable locations at runtime, is that what's happening here?

Not at my desk at the minute but I can do a full review and look into the details later.

Reply:
In current master, "d_fault_inject = 1;" will be executed under the condition of D_FAULT_CONFIG_ENV set and parsed (at Line 605) successfully. Otherwise, "d_fault_inject" will be zero even if under DEV mode. But for developer test, we may not set D_FAULT_CONFIG_ENV, that will cause "d_fault_inject = 1" by default. That is not expected since we need failure inject for development test. I guess that is the reason for Di to make such change in this patch.